pkg/manifests/manifests_test.go

```diff
@@ -1573,6 +1574,136 @@ alertmanagerMain:
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `k8sPrometheusAdapter: `,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/default-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit: {} `,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/default-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit:
+    profile: AllRequestBodies
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/allrequestbodies-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit:
+    profile: WriteRequestBodies
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/writerequestbodies-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+  k8sPrometheusAdapter:
+    audit:
+     profile: None
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/none-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+  k8sPrometheusAdapter:
+    audit:
+      profile: Foobar  # should generate an error
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/none-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: ErrConfigValidation,
+	}}
+
+	for _, test := range tt {
+		c, err := NewConfigFromString(test.config)
+		if err != nil {
+			t.Logf("%s\n\n", test.config)
+			t.Fatal(err)
+		}
+		f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", c,
+			defaultInfrastructureReader(), &fakeProxyReader{}, NewAssets(assetsPath))
+
+		d, err := f.PrometheusAdapterDeployment("foo", map[string]string{
+			"requestheader-allowed-names":        "",
+			"requestheader-extra-headers-prefix": "",
+			"requestheader-group-headers":        "",
+			"requestheader-username-headers":     "",
+		})
+
+		if test.err != nil || err != nil {
+			// fail only if the error isn't what is expected
+			if !errors.Is(err, test.err) {
+				t.Logf("%s\n\n", test.config)
+				t.Fatalf("Expected error %q but got %q", test.err, err)
+			}
+			break
+		}
+
+		adapterArgs := d.Spec.Template.Spec.Containers[0].Args
+		auditArgs := []string{}
+		for _, arg := range adapterArgs {
+			if strings.HasPrefix(arg, "--audit-") {
+				auditArgs = append(auditArgs, arg)
+			}
+		}
+		assertDeepEqual(t, test.expectedArgs, auditArgs,
+			"k8s-prometheus-adapter audit is not configured correctly")
+
+	}
+
+}
+
+func assertDeepEqual(t *testing.T, expected, got interface{}, msg string) {
+	if !reflect.DeepEqual(expected, got) {
```

Comment:
nit:

Is it possible to use `https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff` for better test failure actual vs expected comparison?

Reply:
it is but and I think it is better to use https://pkg.go.dev/gotest.tools but that is a bigger change and this only refactors existing code albeit in new test.

Reply:
Ack